### PR TITLE
refactor(cmd/influxd): move driver code for influxd main package to sub-package so it can be reused

### PR DESCRIPTION
**Pull request by aanthony1243**
- [X] Rebased/mergeable
  - [X] Tests pass

### CODE REVIEW

File: cmd/influxd/launcher/launcher.go [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+	}
+}
+
+func (m *Launcher) Running() bool {
+	return m.running
+}
+
+// URL returns the URL to connect to the HTTP server.
+func (m *Launcher) URL() string {
+	return fmt.Sprintf("http://localhost:%d", m.httpPort)
```
Comment (goller) [inline cmd/influxd/launcher/launcher.go]: This should be 127.0.0.1

Comment (aanthony1243) [inline cmd/influxd/launcher/launcher.go]:
```suggestion
 	return fmt.Sprintf("http://127.0.0.1:%d", m.httpPort)
```

File: cmd/influxd/launcher/launcher_test.go [hunk 2]
```diff
@@ -1,9 +1,10 @@
-package main_test
+package launcher_test
 
 import (
 	"bytes"
 	"context"
 	"fmt"
+	"github.com/influxdata/platform/cmd/influxd/launcher"
```
Comment (jsternberg) [inline cmd/influxd/launcher/launcher_test.go]: Another one here.

File: cmd/influxd/main.go [hunk 3]
```diff
@@ -3,59 +3,27 @@ package main
 import (
 	"context"
 	"fmt"
-	"io"
-	"net"
-	nethttp "net/http"
+	"github.com/influxdata/platform/cmd/influxd/launcher"
```
Comment (jsternberg) [inline cmd/influxd/main.go]: I don't think there's a newline between the stdlib and the library here.

File: go.sum [hunk 4]
```diff
@@ -229,6 +229,7 @@ github.com/influxdata/flux v0.11.0/go.mod h1:crguqnTMQHaGEKp93vZH+pIyTVlJYqkv8bN
 github.com/influxdata/goreleaser v0.86.2-0.20181010170531-0fd209ba67f5/go.mod h1:aVuBpDAT5VtjtUxzvBt8HOd0buzvvk7OX3H2iaviixg=
 github.com/influxdata/influxql v0.0.0-20180925231337-1cbfca8e56b6 h1:CFx+pP90q/qg3spoiZjf8donE4WpAdjeJfPOcoNqkWo=
 github.com/influxdata/influxql v0.0.0-20180925231337-1cbfca8e56b6/go.mod h1:KpVI7okXjK6PRi3Z5B+mtKZli+R1DnZgb3N+tzevNgo=
+github.com/influxdata/kin-openapi v0.1.1-0.20181212221347-ca3615a71f83 h1:z2dPj4rGtqzrwesRBb4yS0+d0a5wcIXiobrhBx1Jm5U=
```
Comment (jsternberg) [inline go.sum]: Is this supposed to be added?

**Pull request merged**